exercises/rail-fence-cipher/example.rs

```diff
@@ -0,0 +1,90 @@
+pub struct RailFence(u32);
+
+fn uncons(s: &str) -> (&str, &str) {
+    s.split_at(match s.chars().next() {
+        Some(c) => c.len_utf8(),
```

Comment:
optional: it seems it would be possible to use https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or here, if it doesn't sacrifice too much readability (in the service of... brevity, I guess?)